sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/model/df/TripleOps.scala

```diff
@@ -0,0 +1,309 @@
+package net.sansa_stack.rdf.spark.model.df
+
+import org.apache.spark.sql._
+import org.apache.spark.rdd.RDD
+import org.apache.jena.graph.{ Triple, NodeFactory }
+import org.apache.spark.sql.types.{ StructField, StructType, StringType }
+
+/**
+ * Spark/DataFrame based implementation of DataFrame of triples.
+ *
+ * @author Gezim Sejdiu
+ */
+
+object TripleOps {
+
+  /**
+   * Convert a [[DataFrame]] into a RDD[Triple].
+   *
+   * @param triples DataFrame of triples (as string).
+   * @return a RDD of triples.
+   */
+  def toRDD(triples: DataFrame): RDD[Triple] = {
+    triples.rdd.map(row =>
+      Triple.create(
+        NodeFactory.createURI(row.getString(0)),
+        NodeFactory.createURI(row.getString(1)),
+        //NodeFactory.createLiteral(row.getString(2))
+        if (row.getString(2).startsWith("http:")) NodeFactory.createURI(row.getString(2)) else NodeFactory.createLiteral(row.getString(2))))
+  }
+
+  /**
+   * Convert a DataFrame of Triple into a Dataset of Triple.
+   *
+   * @param triples DataFrame of triples.
+   * @return a Dataset of triples.
+   */
+  def toDS(triples: DataFrame): Dataset[Triple] = {
+    val spark: SparkSession = SparkSession.builder().getOrCreate()
+    implicit val encoder = Encoders.kryo[Triple]
+    //triples.as[Triple]
+    spark.createDataset[Triple](toRDD(triples))
+  }
+
+  /**
+   * Get triples.
+   *
+   * @param triples DataFrame of triples.
+   * @return DataFrame which contains list of the triples.
+   */
+  def getTriples(triples: DataFrame): DataFrame =
+    triples
```

Comment:
What is the purpose of this function? oO

Reply:
It just return `this` but the user is familiar with the functions provided like `getTriples, getSubjects, getPredicates, getObjects` just to be on the same page! But we can definitely remove it since it contains triple table by default.

Reply:
If you think, that is convenient, then let's keep it.

Reply:
We will keep it for now. But we can change it afterward, if we see it more convenient.